src/main.rs

```diff
@@ -68,13 +50,61 @@ pub async fn inner() -> anyhow::Result<()> {
         .finish();
     tracing::subscriber::set_global_default(subscriber)?;
 
+    // Start a new Figment instance with default values
+    let mut figment =
+        Figment::new().admerge(Serialized::from(config::Config::default(), "default"));
+
+    // Check XDG_CONFIG_HOME for a config file
+    // If XDG_CONFIG_HOME is not set, check HOME for a config file
+    if let Ok(xdg_config_home) = env::var("XDG_CONFIG_HOME") {
+        tracing::info!(
+            "Reading configuration from XDG_CONFIG_HOME: {xdg_config_home:?}/sprocket.toml"
+        );
+        figment = figment.admerge(Toml::file(
+            Path::new(&xdg_config_home.as_str()).join("sprocket.toml"),
+        ));
+    } else if let Some(home) = home_dir() {
+        tracing::info!("Reading configuration from HOME: {home:?}/.config/sprocket.toml");
+        figment = figment.admerge(Toml::file(home.join(".config").join("sprocket.toml")));
+    }
+
+    // Check PWD for a config file
+    if Path::exists(Path::new("sprocket.toml")) {
+        tracing::info!("Reading configuration from PWD/sprocket.toml");
+        figment = figment.admerge(Toml::file("sprocket.toml"));
+    }
+
+    // If provided, check config file from environment
+    if let Ok(config_file) = env::var("SPROCKET_CONFIG") {
+        tracing::info!("Reading configuration from SPROCKET_CONFIG: {config_file:?}");
+        figment = figment.admerge(Toml::file(config_file));
+    }
+
+    // If provided, check command line config file
+    if let Some(ref cli) = cli.config {
+        tracing::info!("Reading configuration from --config: {cli:?}");
+        figment = figment.admerge(Toml::file(cli));
+    }
+
+    // Get the configuration from the Figment
+    let config: Config = figment.extract().expect("Failed to extract config");
+
+    // Write effective configuration to the log
+    tracing::info!(
```

Comment:
I'd prefer not to have such a potentially verbose entry at the `info` level; perhaps `debug`?